metadata-ingestion/scripts/avro_codegen.py

```diff
@@ -559,6 +593,73 @@ def create_from_ids(
 }
 
 
+_extra_schema_methods: Dict[str, List[str]] = {
+    "AuditStampClass": [
+        """
+@staticmethod
+def from_token(
+    auth_token: Optional[str] = None,
+    fallback_actor: Optional[Union["CorpUserUrn", "CorpGroupUrn"]] = None,
+    fallback_timestamp: Optional["datetime"] = None,
+) -> "AuditStampClass":
+    \"\"\"Create an AuditStampClass from a JWT token.
+    
+    This method extracts the actor from the JWT token and creates an AuditStamp.
+    If no token is provided or any error occurs when processing the JWT token, 
+    it defaults to fallback_actor and, if not given, it fallbacks to DEFAULT_ACTOR_URN.
+    
+    Args:
+        auth_token: Optional JWT authentication token.
+        fallback_actor: Optional fallback actor URN.
+        fallback_timestamp: Optional fallback timestamp.
+        
+    Returns:
+        AuditStampClass instance with the appropriate actor and timestamp.
+    \"\"\"
+    import jwt
+    import logging
+    from datetime import datetime
+    from datahub.metadata.urns import CorpUserUrn, CorpGroupUrn
+    
+    logger = logging.getLogger(__name__)
+    DEFAULT_ACTOR_URN = CorpUserUrn("__ingestion")
+    
+    default_actor = fallback_actor or DEFAULT_ACTOR_URN
+    
+    if auth_token is None:
+        actor_urn = default_actor
+    else:
+        try:
+            payload = jwt.decode(auth_token, options={"verify_signature": False})
+        except Exception as e:
+            logger.warning(
```

Comment:
warning feels like it'll get annoying quickly. either we cache this method, or we use some log_once sort of mechanism